drivers/axi_core/spi_engine/spi_engine.c

```diff
@@ -0,0 +1,960 @@
+ * 	will be written
+ * @param reg_data Data that will be written
+ * @return int32_t This function allways returns SUCCESS
+ */
+static int32_t spi_engine_write(struct spi_engine_desc *desc,
+				uint32_t reg_addr,
+				uint32_t reg_data)
+{
+	dev_dbg_reg(desc, "reg_addr = 0x%X; reg_data = 0x%X", reg_addr,
+		    reg_data);
+	axi_io_write(desc->spi_engine_baseaddr, reg_addr, reg_data);
+
+	return SUCCESS;
+}
+
+/**
+ * @brief Read SPI Engine's axi registers
+ *
+ * @param desc Decriptor containing SPI Engine's parameters
+ * @param reg_addr The address of the SPI Engine's axi register from where the
+ * 	data where the data will be read
+ * @param reg_data Pointer where the read that will be stored
+ * @return int32_t This function allways returns SUCCESS
+ */
+static int32_t spi_engine_read(struct spi_engine_desc *desc,
+			       uint32_t reg_addr,
+			       uint32_t *reg_data)
+{
+	dev_dbg_reg(desc, "reg_addr = 0x%X; reg_data = 0x%X", reg_addr,
+		    *reg_data);
+	axi_io_read(desc->spi_engine_baseaddr, reg_addr, reg_data);
+	return SUCCESS;
+}
+
+/**
+ * @brief Write SPI Engine's DMA axi registers
+ *
+ * @param desc Decriptor containing SPI Engine's parameters
+ * @param reg_addr The address of the SPI Engine's DMA axi register where the
+ * 	data will be written
+ * @param reg_data Data that will be written
+ * @return int32_t This function allways returns SUCCESS
+ */
+static int32_t spi_engine_dma_write(struct spi_engine_desc *desc,
+				    uint32_t reg_addr,
+				    uint32_t reg_data)
+{
+	dev_dbg_reg(desc, "reg_addr = 0x%X; reg_data = 0x%X", reg_addr,
+		    reg_data);
+	axi_io_write(desc->tx_dma_baseaddr, reg_addr, reg_data);
+	usleep(100);
+
+	return SUCCESS;
+}
+
+/**
+ * @brief Read SPI Engine's DMA axi registers
+ *
+ * @param desc Decriptor containing SPI Engine's parameters
+ * @param reg_addr The address of the SPI Engine's DMA axi register from where
+ * 	the data where the data will be read
+ * @param reg_data Pointer where the read that will be stored
+ * @return int32_t This function allways returns SUCCESS
+ */
+static int32_t spi_engine_dma_read(struct spi_engine_desc *desc,
+				   uint32_t reg_addr,
+				   uint32_t *reg_data)
+{
+	dev_dbg_reg(desc, "reg_addr = 0x%X; reg_data = 0x%X", reg_addr,
+		    *reg_data);
+	axi_io_read(desc->rx_dma_baseaddr, reg_addr, reg_data);
+
+	return SUCCESS;
+}
+
+/**
+ * @brief Set width of the transfered word over SPI
+ *
+ * @param desc Decriptor containing SPI interface parameters
+ * @param data_wdith The desired data width
+ * 	The supported values are:
+ * 		- 8
+ * 		- 16
+ * 		- 24
+ * 		- 32
+ */
+static int32_t spi_engine_set_transfer_wdith(struct spi_desc *desc,
+		uint8_t data_wdith)
+{
+	struct spi_engine_desc	*desc_extra;
+
+	desc_extra = desc->extra;
+
+	if (data_wdith > desc_extra->max_data_width) {
+		desc_extra->data_width = desc_extra->max_data_width;
+		dev_dbg(desc, "Data width (%d) exceeds the maximum of %d bits",
```

Comment:
Maybe the message here should specifically state that the data width has been changed  to the maximum possible. The user might assume nothing changed based on the error message.